pkg/sql/tests/inverted_index_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tests_test
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
+	"github.com/cockroachdb/cockroach/pkg/util/json"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+const numRandomJSONs = 1000
+const numProbes = 100
+
+func TestInvertedIndex(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{})
+	defer tc.Stopper().Stop(context.TODO())
+
+	db := sqlutils.MakeSQLRunner(tc.Conns[0])
+
+	db.Exec(t, "CREATE DATABASE IF NOT EXISTS test")
+	db.Exec(t, "CREATE TABLE test.jsons (i INT PRIMARY KEY, j JSONB)")
+
+	r := rand.New(rand.NewSource(timeutil.Now().UnixNano()))
+
+	// Grab a bunch of random JSONs. We insert half before we add the inverted
+	// index and half after.
+	jsons := make([]json.JSON, numRandomJSONs)
+	for i := 0; i < numRandomJSONs; i++ {
+		var err error
+		jsons[i], err = json.Random(100, r)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	for i := 0; i < numRandomJSONs/2; i++ {
+		db.Exec(t, `INSERT INTO test.jsons VALUES ($1, $2)`, i, jsons[i].String())
+	}
+	db.Exec(t, `CREATE INVERTED INDEX ON test.jsons (j)`)
+	for i := numRandomJSONs / 2; i < numRandomJSONs; i++ {
+		db.Exec(t, `INSERT INTO test.jsons VALUES ($1, $2)`, i, jsons[i].String())
+	}
+
+	// Now probe it to make sure the data makes sense.
+	for i := 0; i < numProbes; i++ {
+		probeInvertedIndex(t, db, jsons)
+	}
+}
+
+func probeInvertedIndex(t *testing.T, db *sqlutils.SQLRunner, jsons []json.JSON) {
+	j := jsons[rand.Intn(len(jsons))]
```

Comment:
You should use `rand.Perm` in the outer loop instead of this to avoid getting duplicates.

Reply:
Done, PTAL, made the test test a bunch more things.